src/ddsrt/include/dds/config.h.in

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright(c) 2022 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#ifndef DDS_CONFIG_H
+#define DDS_CONFIG_H
+
+#include "dds/version.h"
+#include "dds/features.h"
+
+#cmakedefine DDSRT_WITH_LWIP 1
```

Comment:
Well, yes and no. I've contemplated doing that, but really, ddsrt is not a separate target and there's really only ddsc. The fact that ddsrt can be compiled separately is only because we want to verify system abstractions work without needing to test the logic in ddsi and the dds api's. That's also the reason export headers cannot be generated for it directly. My reasoning is that the config header is a bit like the return codes and time implementations. It's merely a file that's required at the "lowest" level, is there really a point in introducing additional configuration files?

Reply:
I've removed the `features.h` and `version.h` include from `config.h`. So `config.h` can be viewed as the configuration header. The others, especially `features.h`, is the include file that'll tell the user what's available in terms of dds features. As for the location, `features.h` is required by `ddsconf`, so out of practical considerations, I've left them in the same place.